source/adios2/engine/hdf5/HDF5Common.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * HDF5CommonP.h
+ *
+ *  Created on: March 20, 2017
+ *      Author: Junmin
+ */
+
+#ifndef HDF5_COMMON_P_H_
+#define HDF5_COMMON_P_H_
+
+#include "adios2/ADIOSMPICommOnly.h"
+#include "adios2/core/Engine.h"
+
+#include <hdf5.h>
+
+namespace adios
+{
+
+class HDF5Common
+{
+
+public:
+    /**
+     * Constructor for HDF5 file
+     * @param file name
+     */
+    HDF5Common();
+
+    virtual ~HDF5Common();
```

Comment:
No inheritance so no need to make this virtual.  Also can be removed entirely since the implementation is empty and just let the compiler generate a default destructor.